python/adelphi/bin/adelphi

```diff
@@ -131,106 +109,83 @@ def export(ctx, hosts, port, username, password, keyspaces, rf, no_anonymize, ou
-        metadata.update(get_cluster_metadata(cluster))
-        metadata["creation_timestamp"] = datetime.now(utc).isoformat()
-        return (get_keyspaces(props, cluster), {k : metadata[k] for k in metadata.keys() if metadata[k]})
+        return exportclz(cluster, props)
 
-    return with_cluster(build_fn, props)
+    return with_cluster(build_fn, **({k:props[k] for k in ["hosts", "port", "username", "password"]}))
 
 
-def export_keyspaces(props, export_fn, keyspace_map, metadata):
+def export_keyspaces(props, exporter):
     options = {'rf': props['rf']}
     export_dir = props["output-dir"]
 
     if export_dir is None:
-        print(export_fn(iter(keyspace_map.keys()), metadata, options))
+        print(exporter.export_all() if props["include-metadata"] else exporter.export_schema())
     else:
-        for (keyspace_obj, keyspace_id) in keyspace_map.items():
+        def ks_fn(keyspace_obj, keyspace_id):
             # Output filename should use the file-safe alphabet for b64 encoding
-            output_file_name = keyspace_id if props["anonymize"] else keyspace_obj.name
-            with open(os.path.join(export_dir, output_file_name),'w') as output_file:
-                log.info("Writing schema for keyspace " + keyspace_obj.name + " to file " + output_file.name)
+            keyspace_dir_name = keyspace_id if props["anonymize"] else keyspace_obj.name
+            keyspace_dir = os.path.join(export_dir, keyspace_dir_name)
+            if not os.path.exists(keyspace_dir):
+                os.mkdir(keyspace_dir)
+            elif not os.path.isdir(keyspace_dir):
+                log.error("Specified output directory " + keyspace_dir + " exists but is not a directory")
+                exit(4)
+            elif not os.access(keyspace_dir, os.W_OK):
+                log.error("Output directory " + keyspace_dir + " exists but is not writable")
+                exit(5)
+
+            with open(os.path.join(keyspace_dir, "schema"),'w') as schema_file:
+                log.info("Writing schema for keyspace " + keyspace_obj.name + " to file " + schema_file.name)
+                schema_file.write(exporter.export_schema())
+
+            with open(os.path.join(keyspace_dir, "metadata.json"),'w') as metadata_file:
+                log.info("Writing metadata for keyspace " + keyspace_obj.name + " to file " + metadata_file.name)
+                metadata = exporter.export_metadata()
                 metadata['keyspace_id'] = keyspace_id
-                output_file.write(export_fn(iter([keyspace_obj]), metadata, options))
-                del metadata['keyspace_id']
+                metadata_file.write(json.dumps(metadata))
+
+        exporter.each_keyspace(ks_fn)
 
 
 # ============================ Command implementations ============================
 @export.command()
+@click.option('--no-metadata', help="Disable display of metadata when writing to standard out", is_flag=True)
```

Comment:
Added mainly to make the integration test fixes work.  We're adding "creation_timestamp" as a metadata field so unless we want to pull apart the metadata as JSON in the integration test suite we need the ability to ignore timestamps (which will never match a canonical schema).  Making something like jq a part of the test integration framework seemed like overkill here... and all this will be done differently once this framework is moved to Python anyway.

Reply:
I decided _not_ to document this for now.  We already provide an example of how to get help for a specific command in the README so my working assumption was that going from that to "adelphi export-cql --help" wasn't a huge leap.  It's not _abundantly_ clear that this is an option we want to hype up anyway; it certainly could have utility for some users in some cases but I'm not sure we want to encourage it too much.

I'm open to other opinions on this one.